api/src/main/java/com/netflix/iceberg/io/FileIO.java

```diff
@@ -47,4 +46,25 @@
    * Delete the file at the given path.
    */
   void deleteFile(String path);
+
+  /**
+   * Get an {@link OutputFile} to write bytes for a new table metadata file with the given name.
+   */
+  OutputFile newMetadataOutputFile(String fileName);
```

Comment:
Please include `@param filename` that explicitly points out that this should not include a path.